hooks/boot/05-data-helpers.rb

```diff
@@ -0,0 +1,126 @@
+      user_param = 'user'
+      db_param = 'name'
+      param_prefix = 'candlepin_db_'
+    when 'pulpcore'
+      module_name = 'foreman_proxy_content'
+      user_param = 'user'
+      db_param = 'db_name'
+      param_prefix = 'pulpcore_postgresql_'
+    else
+      raise "installer module unknown for db: #{db}"
+    end
+  
+    {
+      host: param_value(module_name, "#{param_prefix}host") || 'localhost',
+      port: param_value(module_name, "#{param_prefix}port") || 5432,
+      database: param_value(module_name, "#{param_prefix}#{db_param}") || db,
+      username: param_value(module_name, "#{param_prefix}#{user_param}"),
+      password: param_value(module_name, "#{param_prefix}password"),
+    }
+  end
+  
+  def pg_command_base(config, command, args)
+    port = "-p #{config[:port]}" if config[:port]
+    "PGPASSWORD='#{config[:password]}' #{command} -U #{config[:username]} -h #{config[:host]} #{port} #{args}"
+  end
+  
+  def pg_sql_statement(config, statement)
+    pg_command_base(config, 'psql', "-d #{config[:database]} -t -c \"" + statement + '"')
+  end
+  
+  # WARNING: deletes all the data from a database. No warnings. No confirmations.
+  def empty_database!(config)
+    generate_delete_statements = pg_sql_statement(config, %q(
+          select string_agg('drop table if exists \"' || tablename || '\" cascade;', '')
+          from pg_tables
+          where schemaname = 'public';
+        ))
+    delete_statements = `#{generate_delete_statements}`
+    execute(pg_sql_statement(config, delete_statements)) if delete_statements
+  end
+  
+  def empty_db_in_postgresql(db)
+    logger.info "Dropping #{db} database!"
+  
+    config = load_db_config(db)
+    if remote_host?(config[:host])
+      empty_database!(config)
+    else
+      execute("sudo -u postgres dropdb #{config[:database]}")
+    end
+  end
+  
+  def reset_candlepin
+    execute('rm -f /var/lib/candlepin/{cpdb_done,cpinit_done}')
+    empty_db_in_postgresql(candlepin)
+  end
+  
+  def load_mongo_config
+    config = {}
+    seeds = param_value('katello', 'pulp_db_seeds')
+    seed = seeds.split(',').first
+    host, port = seed.split(':') if seed
+    config[:host] = host || 'localhost'
+    config[:port] = port || '27017'
+    config[:database] = param_value('katello', 'pulp_db_name') || 'pulp_database'
+    config[:username] = param_value('katello', 'pulp_db_username')
+    config[:password] = param_value('katello', 'pulp_db_password')
+    config[:ssl] = param_value('katello', 'pulp_db_ssl') || false
+    config[:ca_path] = param_value('katello', 'pulp_db_ca_path')
+    config[:ssl_certfile] = param_value('katello', 'pulp_db_ssl_certfile')
+    config
+  end
+  
+  def empty_mongo
+    logger.info "Dropping Pulp database!"
+  
+    mongo_config = load_mongo_config
+    if remote_host?(mongo_config[:host])
+      empty_remote_mongo(mongo_config)
+    else
+      execute(
+        [
+          'systemctl start rh-mongodb34-mongod',
```

Comment:
Should we use `start_services(['rh-mongodb34-mongod'])` here?

Reply:
I suppose if we have a function defined for this exact purpose then we should use it. I've made the change.